kiwi/bootloader/config/systemd_boot.py

```diff
@@ -0,0 +1,159 @@
+#
+import os
+import re
+import glob
+from string import Template
+from typing import Dict
+
+# project
+from kiwi.path import Path
+from kiwi.bootloader.template.systemd_boot import BootLoaderTemplateSystemdBoot
+from kiwi.bootloader.config.bootloader_spec_base import BootLoaderSpecBase
+from kiwi.system.kernel import Kernel
+from kiwi.command import Command
+
+from kiwi.exceptions import (
+    KiwiTemplateError,
+    KiwiBootLoaderTargetError,
+    KiwiKernelLookupError
+)
+
+
+class BootLoaderSystemdBoot(BootLoaderSpecBase):
+    def create_loader_image(self, target: str) -> None:
+        """
+        Handle EFI images
+
+        For systemd boot EFI images are provided along with systemd.
+        Thus the only action taken care of is the creation of the
+        ESP path
+
+        :param str target:
+            target identifier, one of disk, live(iso) or install(iso)
+        """
+        self.efi_boot_path = self.create_efi_path()
+
+    def setup_loader(self, target: str) -> None:
+        """
+        Setup ESP for systemd-boot using bootctl and kernel-install
+
+        For disk images only, setup the ESP layout for systemd boot
+        using bootctl. All data will be loaded from the ESP. The
+        kernel-install script provided with systemd is used to
+        manage kernel updates and the respective loader entries.
+        Distributions integrating well with systemd runs kernel-install
+        as part of their kernel packaging. In kiwi we use kernel-install
+        for the initial boot setup
+
+        :param str target:
+            target identifier, one of disk, live(iso) or install(iso)
+            Currently only the disk identifier is supported
+        """
+        if target != self.target.disk:
+            raise KiwiBootLoaderTargetError(
+                'systemd-boot is only supported with the EFI disk image target'
+            )
+        boot_options = self.custom_args['boot_options']
+        self._mount_system(
+            boot_options.get('root_device'),
+            boot_options.get('boot_device'),
+            boot_options.get('efi_device'),
+            boot_options.get('system_volumes')
+        )
+        kernel_info = Kernel(self.root_mount.mountpoint).get_kernel()
+        if not kernel_info:
+            raise KiwiKernelLookupError(
+                'No kernel in image tree found'
+            )
+        Command.run(
+            [
+                'chroot', self.root_mount.mountpoint, 'bootctl', 'install',
+                '--esp-path=/boot/efi',
+                '--no-variables',
+                '--entry-token', 'os-id'
+            ]
+        )
+        Path.wipe(f'{self.root_mount.mountpoint}/boot/loader')
+        Command.run(
+            [
+                'chroot', self.root_mount.mountpoint,
+                'kernel-install', 'add', kernel_info.version,
+                f'/boot/{kernel_info.name}',
+                f'/boot/initramfs-{kernel_info.version}.img'
```

Comment:
This assumes the kernel and initrd can be found at these locations. Not sure if that's the case for kiwi. In mkosi, we have logic to find the vmlinuz/initrd locations that need to be passed to kernel-install (see https://github.com/systemd/mkosi/blob/1d0492ee5e01bd86916a336c1a2b556c56de0277/mkosi/__init__.py#L3941)

Reply:
I'm pretty sure we have similar logic _somewhere_ else because of Debian and SUSE distributions hotpatching dracut to spit out weird filenames for the images, so we probably should rationalize this with that and provide a utility method to use across everything...

Reply:
the kiwi process makes sure this data exists. some lines up you see an instance of the ```Kernel``` class from which this information is taken. But there is one point that is correct, the name of the initrd varies and I forgot to take the generic interface for it. kiwi also has code to lookup kernels and initrds ;) I'm going to address the initrd part soon

Reply:
done, using the get_boot_names API